main.go

```diff
@@ -100,12 +101,23 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				port := m.list.SelectedItem().FilterValue()
 				m.selectedPort = port
 			} else {
-				// If accepted killing the port, grab PID + execute killPort()
+				// If accepted killing the port, grab PID + get an exec.Cmd for killing a port from killPortCmd()
 				if m.activeButton == "yes" {
-					m.list.ResetFilter()
 					rgx := regexp.MustCompile(`\((.*?)\)`)
 					pid := rgx.FindStringSubmatch(m.list.SelectedItem().FilterValue())[1]
-					killPort(pid)
+
+					killCmd, err := killPortCmd(pid)
+					if err != nil {
+						log.Fatal(err)
+					}
+
+					// Wrap the exec.Cmd in a tea.Cmd and append to cmds []tea.Cmd which will be batched
+					cmds = append(cmds, tea.ExecProcess(killCmd, func(err error) tea.Msg {
+						return err
```

Comment:
I'm pretty sure you can return anything here that you can watch for since `tea.Msg` is a type alias for `interface{}`. So if you wanted to return something specific you could. I didn't think anything was necessary since the desired operation after it succeeds is to restore the process list, which would happen unless `killPortCmd(pid)` returns an error.